great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -1,35 +1,25 @@
 from __future__ import annotations
 
 import copy
-import dataclasses
-import itertools
-from datetime import datetime
-from pprint import pformat as pf
-from typing import (
-    TYPE_CHECKING,
-    Callable,
-    Dict,
-    List,
-    NamedTuple,
-    Optional,
-    Sequence,
-    Type,
-    Union,
-    cast,
-)
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional, Type, cast
 
 import pydantic
-from pydantic import Field
 from pydantic import dataclasses as pydantic_dc
-from typing_extensions import ClassVar, Literal, TypeAlias
+from typing_extensions import ClassVar, Literal
 
 from great_expectations.core.batch_spec import SqlAlchemyDatasourceBatchSpec
-from great_expectations.experimental.datasources.interfaces import (
+
+# The noqa: F401 is to import BatchSorter. This is necessary for pydantic because
```

Comment:
There may be a better way to fix this. Pydantic is implicitly using `BatchSorter`. If I don't import `BatchSorter` I get pydantic errors when running tests (`order_by`, which is of type `List[BatchSorter]` is still a ForwardRef`).  Calling `TableAsset.update_forward_refs()` after defining `TableAsset` fails because `'BatchSorter' is not defined` so this requires the import even if it is not explicitly used. I tried because the error message suggested it, I haven't looked into what is going on. If I import, I do not need to manually call `update_forward_refs`. I haven't dug in anymore and am heading out for the day.

Reply:
Does the `update_forward_refs()` still fail if we do an `if TYPE_CHECKING` import?